sentry-android-core/src/main/java/io/sentry/android/core/UnknownPropertiesTypeAdapterFactory.java

```diff
@@ -104,8 +104,10 @@ public T read(final JsonReader in) {
       }
       // First convert the above JSON tree to an object
       final T object = typeAdapter.fromJsonTree(jsonObjectToParse);
-      // And do the post-processing
-      object.acceptUnknownProperties(unknownProperties);
+      if (!unknownProperties.isEmpty()) {
```

Comment:
@bruno-garcia this will eliminate the `unknown` field error for now as we have all the fields already mapped on our data classes, but deserialization still should be fixed.